R/parse.R

```diff
@@ -154,35 +154,21 @@ coerce_datetime <- function(x, to, ...) {
 # - "2020-01-01T00:02:03-01:00"
 # nolint end
 parse_connect_rfc3339 <- function(x) {
-  # Convert any timestamps with offsets to a format recognized by `strptime`.
+  # Convert timestamps with offsets to a format recognized by `strptime`.
   x <- gsub("([+-]\\d\\d):(\\d\\d)$", "\\1\\2", x)
+  x <- gsub("Z$", "+0000", x)
```

Comment:
I don't understand: these two regexes are not going to both be operating on a timestamp because either it ends in `+00:00` or it ends in `Z` but it can't end in both. Is Connect inconsistent in how it emits timestamps such that we would need both?

Reply:
Connect uses Go's built-in RFC3339 formatter (https://pkg.go.dev/time).

In RFC3339, UTC can be denoted either by `+00:00` or `Z` at the end (https://datatracker.ietf.org/doc/html/rfc3339#section-4.2).

This set of regexes allows us to handle both. (Before this PR we handled by in the `purrr::map_vec()` starting on old line 177.

Reply:
Sure, the spec may allow that, but does Connect actually do it differently in different places? 

This change is defensive against it, so it's fine, I would just be concerned if Connect were internally inconsistent, that seems worth solving so any consumer of the API doesn't have to worry with this fussiness.